sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -69,6 +73,20 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
      * this will be the tracing provider specified in the application context.
      */
     std::shared_ptr<Azure::Core::Tracing::TracerProvider> TracingProvider;
+
+  private:
+    friend class _internal::TelemetryPolicy;
```

Comment:
Is this a friend mainly for testing purposes?

Reply:
This is not for testing - this is so that `TelemetryPolicy` can access the value of `CppStandardVersion` to include it in telemetry string, but no one else could access it, including to set it.

Reply:
In that case, this deserves a comment in source (basically what you shared in the comment is a great answer to the "Why" which would help future maintainers).

We typically have tried to avoid `friend` outside the context of testing. Did we explore ways to do this without `friend`?